src/audio/copier/copier_dai.c

```diff
@@ -64,6 +64,92 @@ static int copier_set_alh_multi_gtw_channel_map(struct comp_dev *dev,
 	return 0;
 }
 
+static int copier_alh_assign_dai_index(struct comp_dev *dev,
+				       void *gtw_cfg_data,
+				       union ipc4_connector_node_id node_id,
+				       struct ipc_config_dai *dai,
+				       int *dai_index,
+				       int *dai_count)
+{
+	struct processing_module *mod = comp_get_drvdata(dev);
+	struct copier_data *cd = module_get_private_data(mod);
+	const struct sof_alh_configuration_blob *alh_blob = gtw_cfg_data;
+	uint8_t *dma_config;
+	size_t alh_cfg_size, dma_config_length;
+	int i, dai_num, ret;
+
+	if (!cd->config.gtw_cfg.config_length) {
+		comp_err(mod->dev, "No gateway config found in blob!");
+		return -EINVAL;
+	}
+
+	switch (dai->type) {
+	case SOF_DAI_INTEL_HDA:
+		/* We use DAI_INTEL_HDA for ACE 2.0 platforms */
```

Comment:
This comment doesn't exactly match with the code, or the connection is not obvious.

What in this piece of code is dependent on ACE2.0?
I don't see any test on the hardware generation....

Reply:
We select `DAI_INTEL_HDA` for LNL and beyond.  For MTL and older platforms we use `DAI_INTEL_ALH`  because they have ALH HW. In line 272 there is a check for `CONFIG_ACE_VERSION_2_0`